evap/staff/tests/test_json_importer.py

```diff
@@ -0,0 +1,202 @@
+
+        importer = JSONImporter(self.semester)
+        importer._import_students(self.students)
+
+        assert UserProfile.objects.all().count() == 2
+
+        user_profile.refresh_from_db()
+
+        self.assertEqual(user_profile.email, self.students[0]["email"])
+        self.assertEqual(user_profile.last_name, self.students[0]["name"])
+        self.assertEqual(user_profile.first_name_given, self.students[0]["christianname"])
+
+    def test_import_lecturers(self):
+        self.assertEqual(UserProfile.objects.all().count(), 0)
+
+        importer = JSONImporter(self.semester)
+        importer._import_lecturers(self.lecturers)
+
+        user_profiles = UserProfile.objects.all()
+        self.assertEqual(user_profiles.count(), 2)
+
+        for i, user_profile in enumerate(user_profiles):
+            self.assertEqual(user_profile.email, self.lecturers[i]["email"])
+            self.assertEqual(user_profile.last_name, self.lecturers[i]["name"])
+            self.assertEqual(user_profile.first_name_given, self.lecturers[i]["christianname"])
+            self.assertEqual(user_profile.title, self.lecturers[i]["titlefront"])
+
+    def test_import_existing_lecturers(self):
+
+        user_profile = baker.make(UserProfile, email=self.lecturers[0]["email"])
+        print(user_profile.email)
+
+        importer = JSONImporter(self.semester)
+        importer._import_lecturers(self.lecturers)
+
+        assert UserProfile.objects.all().count() == 2
+
+        user_profile.refresh_from_db()
+
+        self.assertEqual(user_profile.email, self.lecturers[0]["email"])
+        self.assertEqual(user_profile.last_name, self.lecturers[0]["name"])
+        self.assertEqual(user_profile.first_name_given, self.lecturers[0]["christianname"])
+        self.assertEqual(user_profile.title, self.lecturers[0]["titlefront"])
+
+
+class TestImportEvents(TestCase):
+    def setUp(self):
+        self.semester = baker.make(Semester)
+
+    def _import(self):
+        importer = JSONImporter(self.semester)
+        importer.import_json(EXAMPLE_DATA)
+        return importer
+
+    def test_import_courses(self):
+        importer = self._import()
+
+        self.assertEqual(Course.objects.all().count(), 1)
+        course = Course.objects.all()[0]
+
+        self.assertEqual(course.semester, self.semester)
+        self.assertEqual(course.cms_id, EXAMPLE_DATA["events"][0]["gguid"])
+        self.assertEqual(course.name_de, EXAMPLE_DATA["events"][0]["title"])
+        self.assertEqual(course.name_en, EXAMPLE_DATA["events"][0]["title_en"])
+        self.assertEqual(course.type.name_de, EXAMPLE_DATA["events"][0]["type"])
+        self.assertListEqual(
+            [d.name_de for d in course.degrees.all()], [d["cprid"] for d in EXAMPLE_DATA["events"][0]["courses"]]
+        )
+        self.assertListEqual(
+            list(course.responsibles.all()),
+            [importer.user_profile_map[lecturer["gguid"]] for lecturer in EXAMPLE_DATA["events"][0]["lecturers"]],
```

Comment:
I'd prefer to not have tests rely on the logic that is being tested: If the `user_profile_map` somehow is wrong, the importer would associate the wrong user profiles with the course, but the test wouldn't catch it because it uses the same (wrong) map. I think it's fine if you just hardcode the values we'd expect based on the input data.